homeassistant/components/trello/coordinator.py

```diff
@@ -0,0 +1,87 @@
+"""Data update coordinator for the Trello integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+
+from trello import TrelloClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import LOGGER, Board, List
+
+
+class TrelloDataUpdateCoordinator(DataUpdateCoordinator[dict[str, Board]]):
+    """Data update coordinator for the Trello integration."""
+
+    config_entry: ConfigEntry
+
+    def __init__(
+        self, hass: HomeAssistant, trello_client: TrelloClient, board_ids: list[str]
+    ) -> None:
+        """Initialize the coordinator."""
+        super().__init__(
+            hass=hass,
+            logger=LOGGER,
+            name="trello",
+            update_interval=timedelta(seconds=60),
+        )
+        self.client = trello_client
+        self.board_ids = board_ids
+
+    def _update(self) -> dict[str, Board]:
+        """Fetch data for all sensors using batch API endpoint."""
+        board_lists_query_params = "fields=name"
+        list_cards_query_params = "fields=name&cards=open&card_fields=idCard"
+        urls = ",".join(
+            f"/boards/{board_id}?{board_lists_query_params},/boards/{board_id}/lists?{list_cards_query_params}"
+            for board_id in self.board_ids
+        )
```

Comment:
This looks for me implementation details, which should go into the library

Reply:
I've pulled the implementation details of the batch endpoint into a new Trello lib and refactored them out from this PR. Thanks for pointing this out!